### PR TITLE
[SPARK-25848][SQL][TEST] Refactor CSVBenchmarks to use main method

### PR DESCRIPTION
**Pull request by heary-cao**
## What changes were proposed in this pull request?

use spark-submit:
`bin/spark-submit --class org.apache.spark.sql.execution.datasources.csv.CSVBenchmark --jars ./core/target/spark-core_2.11-3.0.0-SNAPSHOT-tests.jar,./sql/catalyst/target/spark-catalyst_2.11-3.0.0-SNAPSHOT-tests.jar ./sql/core/target/spark-sql_2.11-3.0.0-SNAPSHOT-tests.jar`

Generate benchmark result:
`SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.datasources.csv.CSVBenchmark"`

## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by heary-cao** [comment]
cc @dongjoon-hyun, @wangyum

**Comment by dongjoon-hyun** [comment]
ok to test

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you for your contribution, @heary-cao . 
But, the value seems to be manually copied. Could you run your benchmark actually?
In general, during refactoring, you should rerun the benchmark by yourself and check whether the PR doesn't create some regression.

cc @yucai and @wangyum

**Comment by heary-cao** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVBenchmarks.scala [hunk 1]
```diff
@@ -137,22 +124,15 @@ object CSVBenchmarks extends SQLHelper {
...
-      */
       benchmark.run()
     }
   }
 
-  def main(args: Array[String]): Unit = {
-    quotedValuesBenchmark(rowsNum = 50 * 1000, numIters = 3)
-    multiColumnsBenchmark(rowsNum = 1000 * 1000)
-    countBenchmark(10 * 1000 * 1000)
+  override def runBenchmarkSuite(): Unit = {
```
Comment (yucai) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVBenchmarks.scala]: #22872 has updated `runBenchmarkSuite`'s signature.
```suggestion
  override def runBenchmarkSuite(mainArgs: Array[String]): Unit = {
```

Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVBenchmarks.scala]: +1 for @yucai 's comment.

**Comment by dongjoon-hyun** [comment]
Thank you for updating and rerunning the tests, @heary-cao .

File: sql/core/benchmarks/CSVBenchmarks-results.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+================================================================================================
+Benchmark to measure CSV read/write performance
+================================================================================================
+
+OpenJDK 64-Bit Server VM 1.8.0_163-b01 on Windows 7 6.1
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/CSVBenchmarks-results.txt]: Wow. Did you run this on Windows 7?

**Comment by heary-cao** [comment]
@dongjoon-hyun, Well, my office machine.

File: sql/core/benchmarks/CSVBenchmarks-results.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/CSVBenchmarks-results.txt]: Actually, `GHz` is missing here. So, it's hard to figure out what CPU is used here.
```
Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz [Family 6 Model 94 Stepping 3]
Intel(R) Core(TM) i7-6700T CPU @ 2.80GHz [Family 6 Model 94 Stepping 3]
Intel(R) Core(TM) i5-6600 CPU @ 3.30GHz [Family 6 Model 94 Stepping 3]
Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz [Family 6 Model 94 Stepping 3]
```

Comment (dongjoon-hyun) [reply sql/core/benchmarks/CSVBenchmarks-results.txt]: This seems to be the limitation in Spark benchmark code itself.

Comment (dongjoon-hyun) [reply sql/core/benchmarks/CSVBenchmarks-results.txt]: I made a PR to you. Could you review and merge https://github.com/heary-cao/spark/pull/2 ?

**Comment by dongjoon-hyun** [comment]
@heary-cao . You had better merge in GitHub. GitHub provides merge button for that.

Usually, merging PR is giving a credit to the person.
In Apache Spark, we don't copy someone's code and hide the original authorship.

File: sql/core/benchmarks/CSVBenchmarks-results.txt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+One quoted string                           64733 / 64839          0.0     1294653.1       1.0X
+
+OpenJDK 64-Bit Server VM 1.8.0_191-b12 on Linux 3.10.0-862.3.2.el7.x86_64
+Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz
+Wide rows with 1000 columns:             Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------
+Select 1000 columns                       185609 / 189735          0.0      185608.6       1.0X
+Select 100 columns                          50195 / 51808          0.0       50194.8       3.7X
+Select one column                           39266 / 39293          0.0       39265.6       4.7X
+count()                                     10959 / 11000          0.1       10958.5      16.9X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/CSVBenchmarks-results.txt]: In this case, the ratio change seems to be due to the improvement on `count()`. cc @HyukjinKwon .

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVBenchmarks.scala [hunk 5]
```diff
@@ -16,30 +16,31 @@
...
-object CSVBenchmarks extends SQLHelper {
-  val conf = new SparkConf()
-
-  val spark = SparkSession.builder
-    .master("local[1]")
-    .appName("benchmark-csv-datasource")
-    .config(conf)
-    .getOrCreate()
+
+object CSVBenchmarks extends SqlBasedBenchmark {
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVBenchmarks.scala]: @heary-cao . Could you rename the files?
- `CSVBenchmarks.scala` -> `CSVBenchmark.scala`
- `CSVBenchmarks-results.txt` -> `CSVBenchmark-results.txt`
- [Line 35](https://github.com/apache/spark/pull/22845/files#diff-985fa5181f2aec4df39324995590ea83R35) should be changed together from `benchmarks/CSVBenchmarks-results.txt` to `benchmarks/CSVBenchmark-results.txt`.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Thank you, @heary-cao . Merged to master.

**Comment by heary-cao** [comment]
thanks,@dongjoon-hyum

**Pull request closed without merging**